meps/mep-0003.md

```diff
@@ -0,0 +1,278 @@
+
+## UX implications & migration
+
+The proposed syntax is an addition to existing role and directive patterns. It is a stricly non-breaking enhancement; existing MyST documents that use this syntax are currently considered invalid and should not successfully parse. It does not invalidate or change the behavior of any existing workflows.
+
+The inline attribute syntax adopts the same inline attribute syntax used by Pandoc, Quarto, and djot, with the exception that the inline attribute must start with the role/directive name. This change follows from adapting the reference syntax to MyST Markdown, which already implements extensibilty through nominal types (directives and roles) rather than compositional types (i.e. class names).
+
+## Alternatives Considered
+
+We considered other approaches in this proposal and in discussion over several years. Including:
+
+1. extending the patterns used in Sphinx internal to the role content (`` {name}`content <argument .class1 key="value">` ``)
+2. attributes specified _after_ a role in addition to the name (``{name}`content`{.class1 key="value"}``); and
+
+We chose the current approach to (a) separate the options and content clearly (i.e. not 1), and (b) to keep role names and options close together (i.e. not 2).
+
+### Future Syntax Possibilities
+
+#### Anonymous Roles and Directives
+
+We could provide a short-hand for the div and span roles, where
+
+````
+```{div .red}
+Some markup
+```
+````
+
+becomes
+
+````
+```{.red}
+Some markup
+```
+````
+
+(bool-attrs)=
+
+#### Short-hand Boolean Attibutes
+
+In certain markup languages, such as React JSX, it is possible to pass boolean `true` values simply by name, e.g.
+
+```
+<FooComponent myProp />
+```
+
+where `props.myProp` will be `true`. We might consider extending this syntax such that:
+
+````
+```{code linenos}
+Numbered code directive
+
+Another line
+```
+````
+
+sets the `linenos` option to `true`.
+
+#### Class Coercion
+
+Similarly to [](#bool-attrs), it may be helpful to use the `.CLASS` attribute to set boolean options. Consider the `tip` admonition:
+
+````
+```{tip .dropdown .open}
+```
+````
+
+Where `dropdown` is a valid directive boolean option, setting `.dropdown` could preferentially set `dropdown=true`, and leave the `class` option untouched. This is not a syntax-level enhancement, but rather a proposal that would modify the consumer of the markup e.g. `mystmd`.
+
+## Questions or Objections Considered
+
+Adopting the proposed syntax opens up a currently impossible way to specify options for roles. The same syntax can also be applied to directives, however, it introduces a third way to specify options for directives. We find that acceptable, as this is for short-hand properties like classes and IDs, which ideally should not take up an extra line.
```

Comment:
I think this paragraph sums things up really nicely: Fundamentally, the motivation for this MEP is to enable options, classes, and identifiers on MyST `roles` in a standardized way. This was not possible before and will be possible after this MEP is implemented. 👍

In doing so, it introduces a new, terser way to specify options, classes, and identifiers on `directives` as well. If authors like this new syntax, great, it's less typing for them! If not, no problem. Either way, the structured MyST data will end up the same. I think this follows a nice philosophy of making things for authors as easy and flexible as possible, while not adding bloat at the AST level.